src/syrupy/report.py

```diff
@@ -48,9 +50,23 @@ class SnapshotReport(object):
     updated: "SnapshotFossils" = attr.ib(factory=SnapshotFossils)
     used: "SnapshotFossils" = attr.ib(factory=SnapshotFossils)
 
+    def filter_fossils(self, item: "SnapshotFossil") -> bool:
+        if not self.ran_test_files or not self.is_providing_paths:
+            return True
+
+        target_snaps = {snap.location for snap in self.used}
```

Comment:
`self.used` is populated in https://github.com/tophat/syrupy/pull/123/files#diff-8f899955331827be769212e1ba469cb1R75 after `self.filter_fossils` is used in https://github.com/tophat/syrupy/pull/123/files#diff-8f899955331827be769212e1ba469cb1R67 could be a problem with filtering on incomplete information?

Reply:
Looking into this, might just be able to move this portion below the loop that updates those values to the same effect.

Reply:
Looks like relocating the discovery and filtering below the update to `self.used` has no ill-effect, and I was able to confirm via some savvy prints that the `self.used` is now populated as you'd expect.